File: core/e2e-tests/py/no_business_use.py
"""
File with no business-use imports.
Scanner should skip this file entirely (quick exit).
"""
from flask import Flask

app = Flask(__name__)


@app.route("/health")
def health():
    return {"status": "ok"}


# This function happens to be called "ensure" but it's NOT from business_use
def ensure(condition: bool, message: str) -> None:
    if not condition:
        raise AssertionError(message)


ensure(True, "This should not be extracted")

File: core/e2e-tests/js/no-business-use.ts
/**
 * File with no business-use imports.
 * Scanner should skip this file entirely (quick exit).
 */
import express from 'express';

const app = express();

app.get('/health', (req, res) => {
  res.json({ status: 'ok' });
});

// This function happens to be called "ensure" but it's NOT from business-use
function ensure(condition: boolean, message: string) {
  if (!condition) throw new Error(message);
}

ensure(true, 'This should not be extracted');

export default app;
